### ISSUE TITLE
Cluster controller not working

### ISSUE THREAD

**Issue by taylorsmithgg** [kubecost/cost-analyzer-helm-chart#2416]
**Describe the bug**
This is the only response I'm receiving from the cluster controller:
```
ERR Error unmarshaling request sizing plan in apply handler: unexpected end of JSON input
```

**To Reproduce** 
Configure cloud provider credentials secret

```
helm upgrade --install kubecost \
  --repo https://kubecost.github.io/cost-analyzer/ cost-analyzer \
  --namespace kubecost --create-namespace \
  --set prometheus.nodeExporter.enabled=false \
  --set prometheus.serviceAccounts.nodeExporter.create=false \
  --set prometheus.kubeStateMetrics.enabled=false \
  --set global.prometheus.fqdn=http://prometheus-operated.kube-system.svc:9090 \
  --set global.prometheus.enabled=false \
  --set kubecostToken="<your-token>" \
  --set clusterController.enabled=true \
  --set clusterController.kubescaler.resizeAllDefault=true
```

**Expected behavior**  
The cluster controller should be able to use continuous cluster sizing.

**What impact will this have on your ability to get value out of Kubecost?**
The current implementation of 1-click resizing has the same effect on the latest version of both the cluster controller and kubecost. At the moment we are having to manually apply all changes.

**Bug report**
[kubecost-bug-report.txt](https://github.com/kubecost/cost-analyzer-helm-chart/files/11962688/kubecost-bug-report.txt)

**Comment by kwombach12** [comment]
@kirbsauce Is there anyone on support who can take a look at this?

**Comment by kirbsauce** [comment]
hi @taylorsmithgg !  Sorry you're having issues. To get you to our support team, would you mind emailing support@kubecost.com or posting on our #support channel in Slack?

**Comment by taylorsmithgg** [comment]
@kirbsauce I do not believe there's an active invite link for the slack domain. The current one listed requires @kubecost.com domain. 
<img width="554" alt="image" src="https://github.com/kubecost/cost-analyzer-helm-chart/assets/6597925/d094e02c-f21d-4cc3-aafb-a8c338ab3dd9">

**Comment by kirbsauce** [comment]
Apologies, @taylorsmithgg , can you try this -> https://join.slack.com/t/kubecost/shared_invite/zt-1dz4a0bb4-InvSsHr9SQsT_D5PBle2rw

**Comment by rossfisherkc** [comment]
Received the case; keeping this open for now as it may indeed be a bug.

**Comment by taylorsmithgg** [comment]
@rossfisherkc This should be my primary since it's EKS? Is there a config I'm missing?

**Comment by taylorsmithgg** [comment]
> Apologies, @taylorsmithgg , can you try this -> https://join.slack.com/t/kubecost/shared_invite/zt-1dz4a0bb4-InvSsHr9SQsT_D5PBle2rw

Same result:
<img width="417" alt="image" src="https://github.com/kubecost/cost-analyzer-helm-chart/assets/6597925/b2e4c90b-9efd-4840-ba6d-7571e5c30c27">

**Comment by kirbsauce** [comment]
grrr.  @taylorsmithgg , can you share your email address here and I'll invite you directly?  If not, just email me at kirby@kubecost.com. That said, I do see you created the ticket so we should be moving forward there.

**Comment by kwombach12** [comment]
> @kwombach12 this issue can be converted into a feature request and we should document somewhere of this limitation

I do have a FR for expanding Cluster Controller to the secondary clusters. @bstuder99 Can we add to the docs that cluster controller is only available on the primary cluster today?

**Comment by taylorsmithgg** [comment]
> grrr. @taylorsmithgg , can you share your email address here and I'll invite you directly? If not, just email me at [kirby@kubecost.com](mailto:kirby@kubecost.com). That said, I do see you created the ticket so we should be moving forward there.

The one on the ticket is the correct email

**Comment by bstuder99** [comment]
Ok, the article has been updated to be more clear.

**Comment by rossfisherkc** [comment]
@kwombach12 user confirmed that they are indeed using on a single, primary cluster. I'm glad we still got the doc change out of it.

We can treat this as a bug as it stands, sound good?

**Comment by chipzoller** [comment]
Is this still an issue?

**Comment by Ayush9026** [comment]
I want to solve this issue.

**Comment by grinbvi** [comment]
Did you fix the issue with Cluster Controller?

**Comment by AjayTripathy** [comment]
The cluster controller can be run directly from the secondary, but currently can't be triggered to run from the primary onto the secondary. So I would say the issue still stands but if desired users can install the controller on secondaries and automate resizing via kubectl directly on the secondaries.

**Comment by grinbvi** [comment]
Hi Ajay,

Currently we do not need it.

Thanks,

Vitaly.
